src/atoms/Input/Input.tsx

```diff
@@ -7,34 +7,60 @@ import React, {
 } from 'react';
 import { StyledComponentClass } from 'styled-components';
 
+import Icon, { icons } from 'src/atoms/Icon';
 import styled from 'src/styled-components';
 import Theme, { borderRadius, scale, transitionDuration } from 'src/Theme';
 import { ExtractProps } from 'src/types';
 import Typography from 'src/Typography';
 
-const StyledInput = styled(Typography)`
+const InputContainer = styled.div`
   background: ${props => props.theme.controlBackground};
   border: 0.125em solid ${props => props.theme.controlBorder};
   border-radius: ${borderRadius};
-  font-size: ${scale(0)};
-  font-weight: bold;
-  ${padding(scale(-1), scale(0))};
   transition: border ${transitionDuration}, box-shadow ${transitionDuration};
-
-  :focus {
+  ${padding(scale(-1), scale(0))};
+  display: flex;
+  :focus-within {
     outline: none;
     box-shadow: ${props => props.theme.outline};
   }
+`;
+
+const StyledInput = styled(Typography)<{ iconSide?: string }>`
+  flex: 1;
+  background: none;
+  border: none;
+  font-size: ${scale(0)};
+  font-weight: bold;
```

Comment:
I believe we need this to prevent the second outline from appearing inside the input:
```suggestion
  font-weight: bold;
  outline: none;
```